bin/get-amy.py

```diff
@@ -38,15 +35,25 @@ def main():
     }
 
     # Save.
-    with open(output_file, 'w') as writer:
-        yaml.dump(config, writer, encoding='utf-8', allow_unicode=True)
+    try:
+        with open(output_file, 'w') as writer:
+            yaml.dump(config, writer, encoding='utf-8', allow_unicode=True)
+    except TypeError:
+        # Output_file isn't a string, so it's a file.
+        yaml.dump(config, output_file, encoding='utf-8', allow_unicode=True)
 
 
-def fetch_info(base_url, url):
+def fetch_info(base_url, url, tags=None):
     '''Download and save data.'''
 
     address = base_url + url
 
+    # Consider filtering by tags.
+    if tags:
+        query_params = [urllib.parse.urlencode({'tag': tag}) for tag in tags]
+        query = '&'.join(query_params)
```

Comment:
On Fri, Jan 01, 2016 at 02:01:56PM -0800, Greg Wilson wrote:
> > +        query_params = [urllib.parse.urlencode({'tag': tag}) for tag in tags]
> > +        query = '&'.join(query_params)
>
> Isn't there a utility function in urllib that takes a dict and
> construct the full URL?  Or am I mis-remembering? (Seems odd to have
> to do the '&' and '?' ourselves.

I'd rather not have multiple tag=… entries in one string.  Can't we
do:

  tag=a,b,c

with:

  query = urllib.parse.urlencode({'tag': ','.join(tags)})